test/test_array_data.cpp

```diff
@@ -25,62 +25,60 @@ namespace sparrow
 
         using value_type = std::optional<int>;
         using inner_value_type = value_type::value_type;
+        using inner_reference = int&;
+        using inner_const_reference = const int&;
+        using bitmap_type = array_data::bitmap_type;
+        using bitmap_reference = typename bitmap_type::reference;
+        using bitmap_const_reference = typename bitmap_type::const_reference;
         using reference = reference_proxy<mock_layout>;
         using const_reference = const_reference_proxy<mock_layout>;
         using size_type = std::size_t;
-        using storage_type = std::vector<value_type>;
 
         mock_layout() = default;
         mock_layout(std::initializer_list<value_type> l)
-            : m_storage(l)
+            : m_bitmap(l.size())
+            , m_data(l.size())
```

Comment:
So here we are default-constructing, not sure it's the best way to do this?

Reply:
I cannot think of a better alternative, but I'm opened to any suggestion ;)

Reply:
welp, it's tests so I guess it's fine